diff --git a/Makefile b/Makefile
index 8084ea8..6292d1d 100644
--- a/Makefile
+++ b/Makefile
@@ -505,8 +505,8 @@ $(MUSL_PRINTSCAN_NO_FLOATING_POINT_OBJS): CFLAGS += \
 
 # TODO: apply -mbulk-memory globally, once
 # https://github.com/llvm/llvm-project/issues/52618 is resolved
-$(BULK_MEMORY_OBJS): CFLAGS += \
-        -mbulk-memory
+# $(BULK_MEMORY_OBJS): CFLAGS += \
+#         -mbulk-memory
 
 $(BULK_MEMORY_OBJS): CFLAGS += \
         -DBULK_MEMORY_THRESHOLD=$(BULK_MEMORY_THRESHOLD)
